internal/clientconn/conn.go

```diff
@@ -575,7 +575,8 @@ func (c *conn) route(ctx context.Context, reqHeader *wire.MsgHeader, reqBody wir
 func (c *conn) handleOpMsg(ctx context.Context, msg *wire.OpMsg, command string) (*wire.OpMsg, error) {
 	if cmd, ok := c.h.Commands()[command]; ok {
 		if cmd.Handler != nil {
-			defer observability.FuncCall(ctx)()
+			_, cancel := observability.FuncCall(ctx) // FuncCall's context can't be used for cursors because of cancellation.
```

Comment:
Yeah, I like this idea! Let me try it to see what happens :)

Reply:
That would cause `find` and `aggregate` commands to be unobservable, and that's kind of what we want. I think we should not forget what we are doing and why, not focus only on workarounds…

The simplest solution seem to be to add `observability.FuncCall` calls to each handler. After all, that the function we want to observe, and the reason why we abstract all things inside `FuncCall` – so we can add a single call to each handler, not a lot of calls for tracing/regions/etc